gst/interpipe/gstinterpipesrc.c

```diff
@@ -726,21 +726,19 @@ gst_inter_pipe_src_push_event (GstInterPipeIListener * iface, GstEvent * event,
     ret = gst_pad_push_event (srcpad, event);
   } else {
 
-    event = gst_event_make_writable (event);
     srcbasetime = gst_element_get_base_time (GST_ELEMENT (appsrc));
 
-    if (srcbasetime > basetime) {
-      GST_EVENT_TIMESTAMP (event) =
-          GST_EVENT_TIMESTAMP (event) - (srcbasetime - basetime);
-    } else {
-      GST_EVENT_TIMESTAMP (event) =
-          GST_EVENT_TIMESTAMP (event) + (basetime - srcbasetime);
-    }
```

Comment:
- Out of curiosity, do you know why they were doing this calculation in two different ways?
- Does it matter that we are no longer doing a time correction calculation for events that aren't segments?

Reply:
> * Out of curiosity, do you know why they were doing this calculation in two different ways?

I believe they were avoiding a negative _event timestamp_, but Jan said a negative _segment offset_ is fine.

> * Does it matter that we are no longer doing a time correction calculation for events that aren't segments?

No - apparently that timestamp is never actually used inside gstreamer (they have been meaning to get rid of it). @aidanbarrett that is my recollection, does that align with what you remember?